buildSrc/build.gradle.kts

```diff
@@ -2,12 +2,36 @@ plugins {
     `kotlin-dsl`
 }
 
-repositories {
-    mavenCentral()
-    jcenter()
+/**
+ * These plugins are added to the buildscript classpath, so subprojects can refer to the plugin in the plugins block
+ * by using the [PluginDependenciesSpec.id] function, without the need to also call [PluginDependencySpec.version].
+ */
+val pluginVersions = setOf(
+  "com.jfrog.bintray.gradle:gradle-bintray-plugin:1.8.5",
+  "com.github.jengelman.gradle.plugins:shadow:5.2.0",
+  "com.gradle.publish:plugin-publish-plugin:0.12.0"
+).map {
+    dependencies.implementation(it) ?: throw NullPointerException("Plugin dependency must not be null!")
 }
 
-dependencies {
-    implementation("com.jfrog.bintray.gradle:gradle-bintray-plugin:1.8.4")
-    implementation("com.github.jengelman.gradle.plugins:shadow:2.0.4")
+repositories {
+    exclusiveContent {
+        forRepository {
+            gradlePluginPortal()
+        }
+        filter {
+            pluginVersions.forEach { includeVersion(it.group!!, it.name, it.version!!) }
+        }
+    }
+    jcenter()
+    maven("https://dl.bintray.com/kotlin/kotlin-eap/") {
+        content {
+            includeGroupByRegex("^org\\.jetbrains\\.(anko|dokka|kotlin|kotlinx)(\\..+)?$")
+        }
+    }
+    maven("https://dl.bintray.com/kotlin/kotlin-dev/") {
+        content {
+            includeGroupByRegex("^org\\.jetbrains\\.(anko|dokka|intellij|kotlin|kotlinx)(\\..+)?$")
+        }
+    }
 }
```

Comment:
Here, I added an `exclusiveContent` rule for the plugins to only be taken from `gradlePluginPortal()` and some regex rules for `kotlin-eap` and `kotlin-dev` to avoid unnecessary requests to those repos.

This could be replaced by some more elaborate logic in `buildSrc/src/main/kotlin` that could then be reused in any other Gradle buildscript in the project (except `buildSrc/buildSrc/build.gradle.kts`).